# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [59]:
len(data)

4870

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

## Question 1

### What test is appropriate for this problem? Does CLT apply?

In this exercise, we are investiagating the difference between individuals with black sounding names and white sounding names in regards to receiving a callback from potential employers.

A two-sample t-test is the appropriate choice for this problem since we are comparing two sample means without knowing the true standard deviation of the population.

## Question 2

### What are the null and alternate hypotheses?

Null Hypothesis : Population mean of white names = Population of black names

Alternate Hypothesis : Population mean of white names ≠ Population of black names

## Question 3

### Compute margin of error, confidence interval, and p-value.

In [60]:
b_data = data[data.race == 'b']
len(b_data)

2435

In [61]:
w_data = data[data.race =='w']
len(w_data)

2435

Seperate data into white sounding names and black sounding names

In [48]:
b_calls = b_data.call

In [49]:
w_calls = w_data.call

In [51]:
b_calls.mean(),w_calls.mean()

(0.0644763857126236, 0.09650924056768417)

Find the means of our two sample gorups

In [55]:
diff_calls = w_calls.mean()-b_calls.mean()

In [56]:
diff_calls

0.03203285485506058

There is an approximate 3% difference in the amount of white sound names receive a call as opposed to black sounding names.
Now we will run a t-test to see if that difference is significant enough to reject our null hypothesis.

In [54]:
b_calls.std(),w_calls.std()

(0.24564945697784424, 0.2953455150127411)

In [62]:
len(b_data)+len(w_data)-2

4868

A 95% confidence interval for a two-sample t-test with 4711 degrees of freedom produces a critical t-score of 1.960.

In [66]:
std_err = np.sqrt((w_calls.var()/len(w_calls)) + (b_calls.var()/len(b_calls)))
std_err

0.0077849068268836738

In [67]:
mrg_err = 1.960 * std_err
mrg_err

0.015258417380692

In [70]:
t_stat = diff_calls / std_err
t_stat

4.1147383735462695

In [68]:
stats.ttest_ind(w_calls,b_calls)

Ttest_indResult(statistic=4.1147052908617514, pvalue=3.9408021031288859e-05)

stats.ttest_ind calculates the T-test for the means of two independent samples of scores.

The resulting statistic 4.1147052908617514 is within 0.00003 of the statistic we were able to generate.

##### Results:
* Margin of Error: 0.015258417380692
* Confidence Interval: ± 0.015258417380692
* P-value: 0.0000349

The difference between the two means of our samples is 0.032 which falls above our confidence interval of 0.015.
Therefore, we reject the null hypothesis.

## Question 4
#### Write a story describing the statistical significance in the context or the original problem.

## Question 5

#### Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?